### PR TITLE
Tax logic and parameters from EI_PayrollTax

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
This adds all the variables, parameters and logic from the multi-input, multi-output `EI_PayrollTax` function in `tax-calculator`. It depends on #42 being merged first. 

To handle the multiple outputs, I tried to reduce code duplication as much as possible, and for each returned variable, added the variable-specific logic to its formula; for every intermediate variable whose value was read by multiple output variables, I added a new variable in `intermediate.py`. 

I added all the parameters I could see, and those hard-coded into the policy logic. I'm not 100% sure on the organisation of the parameters, so feel free to suggest any changes.

### CODE REVIEW

File: openfisca_us/parameters/tax/ALD/misc/employer_share.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+description: Employer share of self-employment tax
+values:
+  0000-01-01: 0.5
+metadata:
+  unit: USD
```
Comment (MaxGhenis) [inline openfisca_us/parameters/tax/ALD/misc/employer_share.yaml:5]: not USD (several other parameters are also proportions)

File: openfisca_us/parameters/tax/ALD/misc/self_emp_tax_adj.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+description: Adjustment for self-employment tax haircut
+documentation: This decimal fraction, if greater than zero, reduces the employer equivalent portion of self-employment adjustment. The final adjustment amount would be (1-Haircut)*SelfEmploymentTaxAdjustment.
+values:
+  2013-01-01: 0
+metadata:
+  unit: USD
```
Comment (MaxGhenis) [inline openfisca_us/parameters/tax/ALD/misc/self_emp_tax_adj.yaml:6]: not USD

**Review by MaxGhenis — COMMENTED** [review]
For posterity, taxcalc defines `EIPayrollTax` [here](https://github.com/PSLmodels/Tax-Calculator/blob/2ccfcb446d281aa8f733be59f591de931a41ad24/taxcalc/calcfunctions.py#L102). It returns `sey`, `payrolltax`, `ptax_was`, `setax`, `c03260`, `ptax_oasdi`, `earned`, `earned_p`, `earned_s`, `was_plus_sey_p`, and `was_plus_sey_s`.

This PR adds the following to the `parameters/tax` tree:
* ALD
  * misc
    * employer_share
    * self_emp_tax_adj
* payroll
  * FICA
    * medicare
      * tax_rate
    * social_security
      * add_taxable_earnings
      * max_taxable_earnings
      * tax_rate

This all makes sense to me. NB: There exist non-FICA payroll taxes like unemployment insurance, so `FICA` as a sublevel of `payroll` is appropriate.

It also adds the following variables to the `finance/tax` tree:
* intermediate
  * gross_was_p
  * gross_was_s
  * txearn_was_p
  * txearn_was_s
  * ptax_ss_was_p
  * ptax_ss_was_s
  * ptax_mc_was_p
  * ptax_mc_was_s
  * sey_frac
  * txearn_sey_p
  * txearn_sey_s
  * setax_ss_p
  * setax_ss_p
  * setax_mc_p
  * setax_mc_s
  * setax_p
  * setax_s
  * sey_frac_for_extra_OASDI
  * extra_payrolltax
* outputs
  * sey_p
  * sey_s
  * earned*
  * earned_s*
  * earned_p*
  * was_plus_sey_p
  * was_plus_sey_s
  * payrolltax*
  * sey*
  * c03260*
  * ptax_oasdi*
  * ptax_was*
  * setax*

/* indicates that the variable previously existed, and this PR adds the logic.

Will you be adding tests?

I want to review this a bit more closely. At first glance it looks good but definitely justifies hierarchical calculation to reduce code duplication of payroll tax liabilities between primary and spouse!

File: openfisca_us/parameters/tax/ALD/misc/employer_share.yaml (hunk 1 above)
Comment (nikhilwoodruff) [reply openfisca_us/parameters/tax/ALD/misc/employer_share.yaml]: Done

File: openfisca_us/parameters/tax/ALD/misc/self_emp_tax_adj.yaml (hunk 2 above)
Comment (nikhilwoodruff) [reply openfisca_us/parameters/tax/ALD/misc/self_emp_tax_adj.yaml]: Done

**Comment by nikhilwoodruff** [comment]
Converted taxunit variables to person level variables where appropriate, replacing with `filer_` as discussed.

File: openfisca_us/variables/finance/tax/intermediate.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+class sey_frac(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = u"Taxable fraction of self-employment income"
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        FICA = parameters(period).tax.payroll.FICA
+        sey_frac = 1.0 - 0.5 * (
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/intermediate.py:85]:
```suggestion
        return 1.0 - parameters(period).tax.ALD.misc.employer_share * (
```

File: openfisca_us/variables/finance/tax/intermediate.py [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    value_type = float
+    entity = Person
+    label = u"Taxable self-employment income"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        FICA = parameters(period).tax.payroll.FICA
+        SS = FICA.social_security
+        MC = FICA.medicare
+        txearn_sey_p = min_(
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/intermediate.py:101]:
```suggestion
        return min_(
```

File: openfisca_us/variables/finance/tax/intermediate.py [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+
+class sey_frac_for_extra_OASDI(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = u"Taxable fraction of self-employment income for extra OASDI payroll taxes"
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        FICA = parameters(period).tax.payroll.FICA
+        sey_frac = 1.0 - 0.5 * FICA.social_security.tax_rate
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/intermediate.py:168]:
```suggestion
        return 1.0 - parameters(period).tax.ALD.misc.employer_share * FICA.social_security.tax_rate
```

File: openfisca_us/variables/finance/tax/outputs.py [hunk 6]
```diff
@@ -332,20 +244,96 @@ class sep(Variable):
...
+                    >= STD.aged_or_blind.age_threshold
+                )
+            )
+            * 1
+        )
+        extra_stded = num_extra_stded * STD.aged_or_blind.amount[MARS]
+        return extra_stded
 
 
 class standard(Variable):
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/outputs.py:312]: Is this called `standard` in taxdata? I think `standard_deduction` would be clearer.

File: openfisca_us/variables/finance/tax/outputs.py [hunk 7]
```diff
@@ -332,20 +244,96 @@ class sep(Variable):
...
+    value_type = float
+    entity = TaxUnit
+    label = "Aged and blind standard deduction"
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        STD = parameters(period).tax.deductions.standard
+        MARS = tax_unit("MARS", period)
+        MARSType = MARS.possible_values
+        num_extra_stded = (
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/outputs.py:294]: I'd break this up for clarity
```suggestion
        blind_head = tax_unit("blind_head", period) * 1
        blind_spouse = tax_unit("blind_spouse", period) * 1
        aged_head = (tax_unit("age_head", period) >= STD.aged_or_blind.age_threshold) * 1
        aged_spouse = ((MARS == MARSType.JOINT) & (tax_unit("age_spouse", period) >= STD.aged_or_blind.age_threshold)) * 1
        num_extra_ded = blind_head + blind_spouse + aged_head + aged_spouse
        return num_extra_stded * STD.aged_or_blind.amount[MARS]
```

File: openfisca_us/variables/finance/tax/outputs.py [hunk 8]
```diff
@@ -332,20 +244,96 @@ class sep(Variable):
...
+            tax_unit("blind_head", period) * 1
+            + tax_unit("blind_spouse", period) * 1
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/outputs.py:296]: does the `MARS == MARSType.JOINT` not apply here as it does for aged spouse?

File: openfisca_us/variables/finance/tax/outputs.py [hunk 9]
```diff
@@ -332,20 +244,96 @@ class sep(Variable):
...
+        MIDR = tax_unit("MIDR", period)
+        MARSType = MARS.possible_values
+
+        # Calculate extra standard deduction for aged and blind
+        extra_stded = tax_unit("aged_blind_extra_standard_deduction", period)
+
+        # Calculate the total standard deduction
+        standard = basic_stded + extra_stded
+        standard = where((MARS == MARSType.SEPARATE) & MIDR, 0, standard)
+        standard += charity.allow_nonitemizers * min_(
```
Comment (MaxGhenis) [inline openfisca_us/variables/finance/tax/outputs.py:332]:
```suggestion
        return standard + charity.allow_nonitemizers * min_(
```

**Review by MaxGhenis — COMMENTED** [review]
Could you add this function?
```python
def tax_unit_non_dep_sum(var):
    return tax_unit.sum(
        tax_unit.members(var, period)
        * not_(tax_unit.members("is_tax_unit_dependent", period))
    )
```

File: openfisca_us/variables/finance/tax/outputs.py (hunk 6 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/finance/tax/outputs.py:312]: It is, should've added a comment in the PR. Wasn't sure if we should keep all variable names from tax-calculator when first implementing this?

File: openfisca_us/variables/finance/tax/outputs.py (hunk 8 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/finance/tax/outputs.py:296]: Apparently not: [calcfunctions code](https://github1s.com/pslmodels/tax-calculator/blob/HEAD/taxcalc/calcfunctions.py#L1150)

File: openfisca_us/variables/finance/tax/outputs.py (hunk 9 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/finance/tax/outputs.py]: Done

File: openfisca_us/variables/finance/tax/outputs.py (hunk 7 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/finance/tax/outputs.py]: Done

File: openfisca_us/variables/finance/tax/intermediate.py (hunk 5 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/finance/tax/intermediate.py]: Done

File: openfisca_us/variables/finance/tax/intermediate.py (hunk 4 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/finance/tax/intermediate.py]: dONE

File: openfisca_us/variables/finance/tax/intermediate.py (hunk 3 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/finance/tax/intermediate.py]: Done

**Comment by nikhilwoodruff** [comment]
@MaxGhenis I think I got everything, except about the `standard` variable naming. FWIW I think it'd be better to retain tax-calculator variable names where the variable is at the same level and has the same logic.

File: openfisca_us/variables/finance/tax/outputs.py (hunk 8 above)
Comment (MaxGhenis) [reply openfisca_us/variables/finance/tax/outputs.py]: OK filed https://github.com/PSLmodels/Tax-Calculator/issues/2620 to clarify, but we can revise in a subsequent PR if that turns up a change.

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**